src/data_chain/proof.rs

```diff
@@ -52,6 +53,17 @@ impl Proof {
     }
 }
 
+impl Debug for Proof {
+    fn fmt(&self, formatter: &mut Formatter) -> fmt::Result {
+        write!(
+            formatter,
+            "Proof{{ {}, age: {}, sig: ... }}",
+            self.node_info.name(),
```

Comment:
will it be better to use `PublicInfo`'s Debug trait directly?
I agree the current format is neater, but using `Debug` trait will avoid hassle in case the `PublicInfo` struct got changed?

Reply:
The main motivation was to avoid the signature getting printed, so no, I think we should avoid deriving the `Debug` trait for items which print long (fairly useless) strings of bytes.

Reply:
PublicInfo's Debug trait has been re-implemented and it only contains name & age?
I mean you can have something like:
("Proof{{ {:?}, sig: ... }}", self.node_info)

Reply:
Ah - I see.  That makes it needlessly verbose though since we'd have
```
Proof{ PublicInfo(Node: 7520ea.., age: 55), sig: ... }
```
as opposed to
```
Proof{ 7520ea.., age: 55, sig: ... }
```
Since proofs should only be signed by nodes and not clients, I think it's worth stripping this extra text.

Reply:
Yeah, I agree it's neater to strip the extra text.
Was just thinking using an existing Debug trait will avoid future hassle when PublicInfo struct got changed.
If you think it will not be an issue, I am ok with this change.